### PR TITLE
fix: performance issue on results

### PR DESCRIPTION
**Pull request by chidojiro**
resolve: https://search-io.atlassian.net/jira/software/projects/SF/boards/29?selectedIssue=SF-641

diagnose: rerenders happen so many times that makes the even loop busy and slow to process input events

solution: make items of Results independent from any context

### CODE REVIEW

File: packages/hooks/CHANGELOG.md [hunk 1]
```diff
@@ -199,15 +199,15 @@
   Before:
 
   ```js
-  pipeline.listen('result-clicked', url => {
+  pipeline.listen('result-clicked', (url) => {
```
Comment (zlatanpham) [inline packages/hooks/CHANGELOG.md:202]: I know it's just Eslint stuff but this change is not supposed to be included in the PR.

**Review by zlatanpham — COMMENTED** [review]
This code looks good to me but it's not quite clear how different the performance is after the change. Can we have a before/after comparison or performance profiling evidence for the update?

File: packages/search-ui/src/Results/index.tsx [hunk 2]
```diff
@@ -49,6 +48,9 @@ const Results = (props: ResultsProps) => {
     results,
   ]);
 
+  const { currency, tracking, ratingMax, language } = useSearchUIContext();
```
Comment (wwalser) [inline packages/search-ui/src/Results/index.tsx:51]: Is this the only substantial change, moving the context hook up the tree?

**Review by wwalser — CHANGES_REQUESTED** [review]

**Review by wwalser — DISMISSED** [review]

**Review by zlatanpham — APPROVED** [review]

**Comment by zlatanpham** [comment]
One note is that we are using `JSON.stringify` comparison in [result component](https://github.com/sajari/sdk-react/blob/52ef27ef4c83d769ebc841a856591af887d4b140/packages/search-ui/src/Results/components/Result/index.tsx#L331). This check looks expensive and I'm wondering if we can have an equivalent check by `_id`?

**Comment by zlatanpham** [comment]
> One note is that we are using `JSON.stringify` comparison in [result component](https://github.com/sajari/sdk-react/blob/52ef27ef4c83d769ebc841a856591af887d4b140/packages/search-ui/src/Results/components/Result/index.tsx#L331). This check looks expensive and I'm wondering if we can have an equivalent check by `_id`?

@wwalser just reminded me of one issue that makes my previous assumption incorrect:

> I think the problem with _id in sdk-react is that _id may not always be returned in search queries. It is returned by default, but customers can change the default result fields of their pipeline.

So we are good without touching the part for now.

**Comment by wwalser** [comment]
Rebase on master then feel free to pull the trigger. Thanks @chidojiro

**Review by zlatanpham — APPROVED** [review]

**Pull request merged**